notedown/notedown.py

```diff
@@ -264,37 +264,30 @@ def parse_blocks(self, text):
         code_matches = [m for m in self.code_pattern.finditer(text)]
 
         # determine where the limits of the non code bits are
-        # based on the code block edges
+        # based on the code block edges and save the block
+        # edge to merge code and text blocks.
         text_starts = [0] + [m.end() for m in code_matches]
         text_stops = [m.start() for m in code_matches] + [len(text)]
         text_limits = list(zip(text_starts, text_stops))
 
         # list of the groups from the code blocks
-        code_blocks = [self.new_code_block(**m.groupdict())
+        code_blocks = [(m.start(), self.new_code_block(**m.groupdict()))
                        for m in code_matches]
 
-        text_blocks = [self.new_text_block(content=text[i:j])
+        text_blocks = [(i, self.new_text_block(content=text[i:j]))
                        for i, j in text_limits]
 
         # remove indents
-        list(map(self.pre_process_code_block, code_blocks))
+        for pos, block in code_blocks:
```

Comment:
Using a for-loop explicitly states that we are side-effecting and we're not interested in the map return.